.github/workflows/magma-build-3rd-party.yml

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Magma Build & Publish 3rd Party Dependencies
+on:
+  push:
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/**
+      - .github/workflows/magma-build-3rd-party.yml
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/**
+      - .github/workflows/magma-build-3rd-party.yml
+  schedule:
+    - cron: 0 4 * * 0
+  workflow_dispatch:
+    inputs:
+      production:
+        description: Is production build?
+        type: boolean
+        default: false
+
+jobs:
+  build_dependencies:
+    runs-on: ubuntu-latest
+    container: ghcr.io/magma/magma/devcontainer:latest
+    env:
+      repository: debian-ci
+
+    strategy:
+      fail-fast: false
+      matrix:
+        distribution: [ focal ]
```

Comment:
> is the distribution not focal-ci?

I'd say: no, with the following thoughts. However, I can also change, if you feel there is a reason.
- The current repository https://artifactory.magmacore.org/ui/repos/tree/General/debian-prod lists `focal` and `stretch` (amongst others) but no `focal-ci`.
- The actual distribution in Ubuntu is called `focal` and not `focal-ci`. (Yes, I also think, that the `focal-1.x.y` is rubbish and a mistake. There are version numbers on packages for that...)
- Calling it `*-ci` would be a unnecessary duplication, as the repository is already called `*-ci`.

> why use a matrix argument if the value is always the same?

- Future perspective: We might want to build for other distributions as well. We would already have the structure in place than to do so easily.
- Convenience: All relevant parameters for the parameter-space are configure consistently in one place.
- "Good looks": The automatic job naming is regarding this, and I found it rather nice.

Reply:
Even if there are other approaches to organize the distributions, I do not think it makes sense to only change this for the third party use case. This would need changes everywhere the repo is wired in VMs, Containers and even production setups.

If we decide to change this, it must be discussed and planned.

Reply:
> - The current repository https://artifactory.magmacore.org/ui/repos/tree/General/debian-prod lists `focal` and `stretch` (amongst others) but no `focal-ci`.

So this is no change imho, but the continuance of the current string.

Reply:
We are publishing all development artifacts to debian-test focal-ci - debian-prod is filled by copying the artifacts over during a release.

Reply:
So this is messy. Defaulted to focal-ci now.